client/src/components/ScatterChart.stories.tsx

```diff
@@ -0,0 +1,127 @@
+import { BLUE, GREEN, GREY, PURPLE } from 'components/charts/colours';
+import React from 'react';
+import ScatterChart, { AxisProps, DataProps } from './ScatterChart';
+
+export default {
+    component: ScatterChart,
+    title: 'ScatterChart'
+};
+
+interface ChartProps {
+    title: string,
+    xAxis: AxisProps,
+    yAxis: AxisProps,
+    dataColour: string,
+    maxColour: string
+}
+
+interface DataArgs {
+    data: DataProps[],
+    max: number
+}
+
+type Template = ((a: DataArgs) => JSX.Element) & { args: DataArgs };
+
+const createTemplate = (chartArgs: ChartProps): Template =>
+    Object.assign(
+        // Template function
+        (a: DataArgs) => <ScatterChart {...{ ...chartArgs, ...a }} />,
+
+        // This lets us use the args prop later on
+        { args: {} as DataArgs }
+    );
+
+// This is used to create templates for each of the scatter based charts
+const Templates = {
+    VelocityDistance: createTemplate({
+        title: "Velocity-Distance",
+        xAxis: { label: 'Distance', unit: 'km' },
+        yAxis: { label: 'Velocity', unit: 'km/hr' },
+        dataColour: PURPLE,
+        maxColour: GREY
+    }),
+    PowerDistance: createTemplate({
+        title: "Power-Time",
+        xAxis: { label: 'Distance', unit: 'km' },
+        yAxis: { label: 'Power', unit: 'W' },
+        dataColour: BLUE,
+        maxColour: GREY
+    }),
+    CadenceDistance: createTemplate({
+        title: "Cadence-Distance",
+        xAxis: { label: 'Distance', unit: 'km' },
+        yAxis: { label: 'Cadence', unit: 'RPM' },
+        dataColour: GREEN,
+        maxColour: GREY
+    }),
+};
+
+type Point = { x: number, y: number }
+
+// Quadratic scaled to a maximum value
+const quad = (low: number, high: number, max: number) =>
+    (x: number) => -((x - low) * (x - high)) * max / (1 / 4 * (low - high) * (low - high));
+
+// Add random noise to a value
+const perturb = (n: number) => Math.random() > 0.75
+    ? n : Math.abs(n * (Math.random() / 10 + 0.95) + (Math.random() - 0.5) * 14);
```

Comment:
I would prefer that our stories were deterministic. Feel free to add some noise, either hardcode it or use something deterministic (another algorithm or see if you can give JS a fixed seed)

This not only means it renders the same for us every time, but also means if we ever add some automated testing like in the storybook tutorial things won't break.

Reply:
Done
7aef92c